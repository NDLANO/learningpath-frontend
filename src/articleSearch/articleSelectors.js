/**
 * Copyright (c) 2016-present, NDLA.
 *
 * This source code is licensed under the GPLv3 license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import { createSelector } from "reselect";
import { convertFieldWithFallback } from "../util/convertFieldWithFallback";

const getTotalResultsFromState = (state) => {
  const { result } = state.embedSearch.ndla;
  return result.totalCount ? result.totalCount / result.pageSize : 0;
};

export const getNumberOfArticlePages = createSelector([getTotalResultsFromState], (totalResults) =>
  totalResults ? Math.ceil(totalResults / 10) : 1,
);

export const getArticleResultFromState = (state) => {
  const results =
    state.embedSearch.ndla.result && state.embedSearch.ndla.result.results ? state.embedSearch.ndla.result.results : [];
  return results.map((item) => ({
    ...item,
    introduction: convertFieldWithFallback(item, "introduction", ""),
    title: convertFieldWithFallback(item, "title", ""),
    thumbnail: item.metaImage && item.metaImage.url ? item.metaImage.url : `/placeholder.png`,
    thumbnailAlt:
      item.metaImage && item.metaImage.alt ? item.metaImage.alt : convertFieldWithFallback(item, "title", ""),
    showUrl: false,
  }));
};
