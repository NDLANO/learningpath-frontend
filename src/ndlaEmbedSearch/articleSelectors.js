/**
 * Copyright (c) 2016-present, NDLA.
 *
 * This source code is licensed under the GPLv3 license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import { createSelector } from 'reselect';
import requireAssets from '../util/requireAssets';
import { convertFieldWithFallback } from '../util/convertFieldWithFallback';

const getTotalResultsFromState = (state) => {
  const result = state.embedSearch.ndla.result;
  return result.totalCount ? result.totalCount / result.pageSize : 0;
};

export const getNumberOfArticlePages = createSelector(
  [getTotalResultsFromState],
  totalResults => totalResults ? Math.ceil(totalResults / 10) : 1,
);

export const getArticleResultFromState = (state) => {
  const results = state.embedSearch.ndla.result && state.embedSearch.ndla.result.results ? state.embedSearch.ndla.result.results : [];
  return results.map((item) => ({
    id: item.id,
    description: convertFieldWithFallback(item, 'description', ''),
    title: convertFieldWithFallback(item, 'title', ''),
    link: `https://ndla-frontend.test.api.ndla.no/article/${item.id}`,
    thumbnail: `/assets/${requireAssets['placeholder.png']}`,
  }));
}

export const getEmbedQueryFromState = (state, type) =>
  state.embedSearch.ndla.query;

export const getOembedContentFromState = (state, type) =>
  state.embedSearch[type].oembedContent;
