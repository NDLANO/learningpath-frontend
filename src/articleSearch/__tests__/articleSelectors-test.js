/**
 * Copyright (c) 2016-present, NDLA.
 *
 * This source code is licensed under the GPLv3 license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import { getArticleResultFromState, getNumberOfArticlePages } from "../articleSelectors";

const state = {
  embedSearch: {
    ndla: {
      result: {
        results: [
          {
            id: 1,
            title: { title: "Test", language: "nb" },
            introduction: { introduction: "Test", language: "nb" },
            disabled: false,
          },
          {
            id: 2,
            title: { title: "Test3", language: "nb" },
            introduction: { introduction: "Test3", language: "nb" },
            disabled: false,
          },
          {
            id: 3,
            title: { title: "Test2", language: "nb" },
            introduction: { introduction: "Test2", language: "nb" },
            disabled: false,
          },
        ],
        totalCount: 3,
        pageSize: 10,
      },
    },
  },
};

test("selectors/getArticleResultFromState", () => {
  expect(getArticleResultFromState(state)).toMatchSnapshot();
});

test("selectors/getNumberOfArticlePages", () => {
  expect(getNumberOfArticlePages(state)).toEqual(1);
});
