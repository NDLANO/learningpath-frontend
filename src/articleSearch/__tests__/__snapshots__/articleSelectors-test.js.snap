// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`selectors/getArticleResultFromState 1`] = `
Array [
  Object {
    "disabled": false,
    "id": 1,
    "introduction": "Test",
    "showUrl": false,
    "thumbnail": "/placeholder.png",
    "title": "Test",
  },
  Object {
    "disabled": false,
    "id": 2,
    "introduction": "Test3",
    "showUrl": false,
    "thumbnail": "/placeholder.png",
    "title": "Test3",
  },
  Object {
    "disabled": false,
    "id": 3,
    "introduction": "Test2",
    "showUrl": false,
    "thumbnail": "/placeholder.png",
    "title": "Test2",
  },
]
`;
